### DISCUSSION TITLE
LDAP Active Directory Group Help

### DISCUSSION THREAD

**Discussion by randombdk** [ngoduykhanh/PowerDNS-Admin#1525] [Q&A]
Hi,

So we have been running powerdns admin for a while :) but after upgrade of the docker container to latest 0.4.1 i seem to have problem i thought only was in the old version

if i set ldap_sg_enabled = False i can login with my ad user, if i enabled them and add the respective ad groups based on distinguished names it seems to fail:

[2023-04-13 09:58:14 +0000] [8] [ERROR] User 'xxx' does not belong to any group while LDAP_GROUP_SECURITY_ENABLED is ON
[2023-04-13 09:58:14,746] [user.py:305] ERROR - User 'xxx' does not belong to any group while LDAP_GROUP_SECURITY_ENABLED is ON
[2023-04-13 09:58:14 +0000] [8] [WARNING] User xxx failed to authenticate via LDAP from 192.168.241.3
[2023-04-13 09:58:14,746] [index.py:579] WARNING - User lkn failed to authenticate via LDAP from 192.168.241.3

This seems to pop aswell: LdapErr: DSID-0C090A58, comment: In order to perform this operation a successful bind must be completed on the connection.

Is there a easy fix for this? because if i set to disable it works..

**Comment by AzorianMatt** [comment]
@randombdk what is the last version you were running where you know this worked?

**Reply by randombdk** [reply]
Hi Matt, that would be latest prior upgrading to 0.4.1 - i dont remember the exact version

But if i just disable the security group - i can still ldaps validate ? :D .. i dont get that?

**Reply by randombdk** [reply]
i think it was 0.3.7 if im not wrong

**Reply by AzorianMatt** [reply]
@randombdk `0.4.0` came right before `0.4.1` but there was certainly never a `0.3.7` or even a `0.2.7` for that matter.

As for the why, I'm sure there is just an oversight in the logic as this app tends to have that scenario come up enough.

I took a look at the code and it seems if you turn on debug mode, then you might get some useful info here. It would appear you're getting that message because there is a scenario where no results are coming back from the LDAP server. I'm not entirely certain on this though as I would need to study the code a bit more in depth.

**Reply by randombdk** [reply]
ahh - okey that makes sense i thought if i disable the security groups check i shouldent be able to login

I will try check which version i was on prior

But i think it was the public docker container (not sure on the version)

**Reply by AzorianMatt** [reply]
If you can track down what container image you were using, that should give away the version. This will better help me determine where the issue came from and thus, what changes to look for.

Thanks!

**Comment by randombdk** [comment]
Sure thing

I will check back tomorrow on work 😊

**Comment by randombdk** [comment]
![Uden titel](https://user-images.githubusercontent.com/105320376/231965842-7273e0a8-2558-41d7-a093-1b7a2d4893ea.png)
Hi Matt

ive restored the server for a few days ago

i dont think i get closer, it was latest version before 0.4 was released - so i think you might now what image it was better than me :-)


i hope this helps.

Den 13. april 2023 kl. 19.42.56 +02.00, skrev Matt Scott ***@***.***>:

**Comment by glomazdov** [comment]
I have the same problem but not in docker
ERROR - User 'ХХХХ' does not belong to any group while LDAP_GROUP_SECURITY_ENABLED is ON
in version 0.4.0 Everything worked, broke in 0.4.1
AND
missing fields
LDAP admin username 
LDAP admin password

**Reply by randombdk** [reply]
Have you tried reversing the image to see if its fine again ? if you remove the 0.4.1?

**Reply by randombdk** [reply]
It still works if you manually se ldap_sg_group = False (atleast you can still login) in the database.

**Reply by AzorianMatt** [reply]
@glomazdov why do you say those two fields are missing? They were only shown for openLDAP before which is what I carried over with the overhaul. Ergo, if you're all using the Windows AD LDAP type, you would not see those fields.

What am I missing here?

**Reply by glomazdov** [reply]
yes, I was wrong.

**Reply by AzorianMatt** [reply]
@glomazdov should those settings be present for an AD environment or no? I have no clue as I just carried over what was already there.

**Comment by glomazdov** [comment]
> Hi,
> 
> So we have been running powerdns admin for a while :) but after upgrade of the docker container to latest 0.4.1 i seem to have problem i thought only was in the old version
> 
> if i set ldap_sg_enabled = False i can login with my ad user, if i enabled them and add the respective ad groups based on distinguished names it seems to fail:
> 
> [2023-04-13 09:58:14 +0000] [8] [ERROR] User 'xxx' does not belong to any group while LDAP_GROUP_SECURITY_ENABLED is ON [2023-04-13 09:58:14,746] [user.py:305] ERROR - User 'xxx' does not belong to any group while LDAP_GROUP_SECURITY_ENABLED is ON [2023-04-13 09:58:14 +0000] [8] [WARNING] User xxx failed to authenticate via LDAP from 192.168.241.3 [2023-04-13 09:58:14,746] [index.py:579] WARNING - User lkn failed to authenticate via LDAP from 192.168.241.3
> 
> This seems to pop aswell: LdapErr: DSID-0C090A58, comment: In order to perform this operation a successful bind must be completed on the connection.
> 
> Is there a easy fix for this? because if i set to disable it works..

did you find a solution to the problem?

**Reply by randombdk** [reply]
for now ive disabled the check in the database :) and awaiting a fix
